### ISSUE TITLE
Takes forever to connect to matrix

### ISSUE THREAD

**Issue by ryzokuken** [torhve/weechat-matrix-protocol-script#83]
Hi all!

For me, it takes forever to connect to matrix. In the past, I have connected a few times, but usually I grow impatient. Maybe something's wrong with my configuration? Is it usual for it to take too long to connect?

Weechat version: 1.9

**Comment by wucke13** [comment]
What do you mean that it takes forever? Normally, it should take five to seven seconds until `Call to API errored in command url:https://matrix.org/_matrix/client/r0/login, maybe timeout?` shows up in the matrix main view, which of course means that the connect did not succeed. Does this occur?

**Comment by ryzokuken** [comment]
@wucke13 Yes, I usually get this kind of messages. How long does the plugin wait after such an error before sending another request?

**Comment by wucke13** [comment]
@ryzokuken In the actual build there is a hardcoded timeout, which happens to be 5 seconds in most cases. I made some modifications, which allow you to specify a custom timeout via the weechat settings interface. This change is still waiting to be merged into the main branch. If you want to try whether this fixes your problem, feel free to try my changed version from the forked repo in my github account.

**Comment by ryzokuken** [comment]
Okay. I will try building weechat from source.

**Comment by wucke13** [comment]
No, there is no need to do so. I expressed myself in a misleading manner. The "hardcoded timeout" is in `matrix.lua`. There is no need to change anything in weechat itself, so no need to recompile it. Just exchange your `matrix.lua` with the one from my github repo.

**Comment by ryzokuken** [comment]
I'll add your repo as a remote and shift to its master, thanks.
